mypy/binder.py

```diff
@@ -210,10 +220,19 @@ def pop_frame(self, can_skip: bool, fall_through: int) -> Frame:
 
         return result
 
+    @contextmanager
+    def accumulate_type_assignments(self) -> 'Iterator[Assigns]':
+        self.type_assignments = defaultdict(list)
```

Comment:
Add docstring. Also mention that this is used for multi-assignment from union (and why this is needed there).